src/core/shape_inference/include/one_hot_shape_inference.hpp

```diff
@@ -67,7 +67,7 @@ std::vector<TRShape> shape_infer(const OneHot* op,
         auto depth_as_shape =
             get_input_const_data_as_shape<TRShape>(op, 1, ta, util::GetNotNegative<typename DimType::value_type>(op));
 
-        if (depth_as_shape && depth_as_shape->size() == 1) {
+        if (depth_as_shape && depth_as_shape->size() == 1 && (*depth_as_shape)[0].get_length() > 0) {
```

Comment:
I'm confused as to why this change in shape inference is necessary. Per the specification, depth is a positive scalar. If it is == 0, then this is an error per specification. Can you accept depth as an input in GPU?

Reply:
@jane-intel depth is default to 0 in one_hot primitive:
```
int64_t depth = 0; 
```
Previously it hopes to get the depth value from the second input (Constant node) of OneHot node, and then use that depth value to create one_hot primitive. But in model (in test_stable-fast-3d.ipynb), the second input of OneHot node is Select node instead of Constant node, so no depth value can be obtained during compile_model. 

```
If it is == 0, then this is an error per specification.
```
For this abnormal model, I leave the depth to be the default 0 and not a positive scalar. 

The shape_infer of one_hot will get output shape [?,0]. It will then be failed at shape_infer of the following ScatterNDUpdate node.
![image](https://github.com/user-attachments/assets/554b303d-6225-41bb-914c-182ec48dc95d)
Because the updates_shape[indices_rank-1:] is 0 (updates_shape = [?, 0], which is got from shape_infer with depth = 0), and input_shape[indices_shape[-1]:] is 2 in this case, and they are not compatible. 
```
                NODE_VALIDATION_CHECK(
                    op,
                    std::all_of(inputs_shape.begin() + last_idx_dim_size, inputs_shape.end(), is_update_compatible),
                    "updates_shape[indices_rank-1:] shape must be input_shape[indices_shape[-1]:]");
```
So I changed the mentioned line code to set the updates_shape to "[?,?]" instead of "[?, 0]", and it can be compatible to 2 (input_shape[indices_shape[-1]:]) now.

Reply:
I don't see this model as abnormal. Depth is not guaranteed to be constant.
The fact that depth defaults to 0 in the plugin -- is an implementation detail of the plugin and plugin only. I don't see why this should affect the way common shape inference works for this operation. Especially because 0 is not a permissible value by spec.

Reply:
@yuanxion, it seems we just should pass tensor_accessor with empty tensors map, which will result in a dynamic dimension for the output shape in cases like this when the actual depth value is not known yet at the moment of the shape inference call. Could you please try this approach while keeping the shape_infer function itself unchanged?

Reply:
Thanks @jane-intel  & @sshlyapn. After using empty const_data map and keeping shape_infer as it is (commit 
82b1706), it can get dynamic output_shape during compile-time. And the final 3D output result is almost the same as CPU as well.
![image](https://github.com/user-attachments/assets/ba566fdf-9205-4f8a-96e8-5666dc6cd1c4)